File: SantanderTeste/app/src/main/java/com/example/alessandrofsouza/santanderapp/presentation/adapters/PageAdapter.java
package com.example.alessandrofsouza.santanderapp.presentation.adapters;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.example.alessandrofsouza.santanderapp.R;
import com.example.alessandrofsouza.santanderapp.presentation.pages.contact.ContactFragment;
import com.example.alessandrofsouza.santanderapp.presentation.pages.contact.ContactPresenter;
import com.example.alessandrofsouza.santanderapp.presentation.pages.investment.InvestmentFragment;
import com.example.alessandrofsouza.santanderapp.presentation.pages.investment.InvestmentPresenter;

import java.util.ArrayList;

public class PageAdapter extends FragmentPagerAdapter {

    private ArrayList<Fragment> fragments = new ArrayList<>();
    private ArrayList<String> title = new ArrayList<>();
    private Context context;

    public PageAdapter(Context ctx, FragmentManager fm) {
        super(fm);
        context = ctx;
    }

//    public void add(Fragment frag, String title) {
//        this.fragments.add(frag);
//        this.title.add(title);
//    }

    @Override
    public Fragment getItem(int i) {
//        return this.fragments.get(i);

        switch (i) {
            case 0:
                InvestmentFragment fragment1 = InvestmentFragment.newInstance();
                new InvestmentPresenter(fragment1);
                return fragment1;

            case 1:
                ContactFragment fragment0 = ContactFragment.newInstance();
                new ContactPresenter(fragment0);
                return fragment0;

            default:
                return null;

        }
    }

    @Override
    public int getCount() {
//        return this.fragments.size();
        return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {
//        return this.title.get(position);
        switch (position) {
            case 0:
                return context.getResources().getString(R.string.investimento);
            case 1:
                return context.getResources().getString(R.string.contato);
            default:
                return null;
        }
    }
}

File: app/src/main/java/com/anabhomasi/androidapp/views/adapters/PageAdapter.kt
package com.anabhomasi.androidapp.views.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import com.anabhomasi.androidapp.views.fragments.DynamicFormFragment
import com.anabhomasi.androidapp.views.fragments.FundFragment
import com.anabhomasi.androidapp.views.fragments.SuccessFormFragment

class PageAdapter(fragmentManager: FragmentManager) :
        FragmentPagerAdapter(fragmentManager) {

    // Returns total number of pages
    override fun getCount(): Int {
        return NUM_ITEMS
    }

    // Returns the fragment to display for that page
    override fun getItem(position: Int): Fragment? {
        return when (position) {
            FUND_PAGE -> FundFragment.newInstance("Investimento", FUND_PAGE)
            FORM_PAGE -> DynamicFormFragment.newInstance()
            SUCCESS_FORM_PAGE -> SuccessFormFragment.newInstance("Contato", SUCCESS_FORM_PAGE)
            else -> null
        }
    }

    // Returns the page title for the top indicator
    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            FUND_PAGE -> "Investimento"
            FORM_PAGE, SUCCESS_FORM_PAGE -> "Contato"
            else -> ""
        }
    }

    companion object {
        private const val NUM_ITEMS = 3

        const val FUND_PAGE = 0
        const val FORM_PAGE = 1
        const val SUCCESS_FORM_PAGE = 2
    }
}